### PR TITLE
chore: add contributing file

### LINKED ISSUE #59: [Chore]: Create a CONTRIBUTING document
### What package(s) should this feature be added to?

_No response_

### Summary

# Description

Create a CONTRIBUTING markdown file similar to the [FAST CONTRIBUTING markdown file](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md). Editing where necessary to be FAST tooling focused.

### PR DESCRIPTION
**Pull request by williamw2**
# Pull Request

## 📖 Description

Adding a contributing.md file for the repo.

### 🎫 Issues
closes #59 

### General

- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
+# Contributing to FAST Tooling
+
+## Getting started
+
+### Machine setup
+
+To work with the FAST Tooling [monorepo](https://en.wikipedia.org/wiki/Monorepo) you'll need Git, Node.js, Yarn and Lerna setup on your machine.
```
Comment (janechu) [inline CONTRIBUTING.md:7]:
```suggestion
To work with the FAST Tooling [monorepo](https://en.wikipedia.org/wiki/Monorepo) you'll need Git, Node.js, and NPM setup on your machine.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+```shell
+npm install -g yarn
+```
+
+The second tool you'll need is Lerna, which can be installed with this command:
+
+```bash
+yarn global add lerna
+```
+
```
Comment (janechu) [inline CONTRIBUTING.md:24]:
```suggestion
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+```shell
+git clone git@github.com:microsoft/fast-tooling.git
+```
+
+### Installing and building
+
+From within the `fast-tooling` folder where you've cloned the repo, install all package dependencies and build all workspaces (local dependencies) with this command:
+
+```bash
+yarn
```
Comment (janechu) [inline CONTRIBUTING.md:47]:
```suggestion
npm install
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+```
+
+After the initial install, you can re-build all workspaces in the future with:
+
+```bash
+yarn prepare
```
Comment (janechu) [inline CONTRIBUTING.md:53]:
```suggestion
npm run prepare --workspaces
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+```
+
+### Testing
+
+To run all tests for all packages, use the following command:
+
+```bash
+yarn test
```
Comment (janechu) [inline CONTRIBUTING.md:61]:
```suggestion
npm run test --workspaces
```

**Review by EisenbergEffect — APPROVED** [review]

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+
+This command can also be run from within individual package folders to execute only tests from that package.
+
+:::note
+Packages are located within the `packages` folder of the repository. Each package has a `package.json` file with a `scripts` section that defines the commands available to you for common tasks such as build, test, lint, etc.
+:::
+
+### Submitting a pull request
+
+If you'd like to contribute by fixing a bug, implementing a feature, or even correcting typos in our documentation, you'll want to submit a pull request. Before submitting a pull request, be sure to [rebase](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) your branch from master or use the *merge* button provided by GitHub.
```
Comment (janechu) [inline CONTRIBUTING.md:72]:
```suggestion
If you'd like to contribute by fixing a bug, implementing a feature, or even correcting typos in our documentation, you'll want to submit a pull request. Before submitting a pull request, be sure to [rebase](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) your branch from main or use the *merge* button provided by GitHub.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+When submitting your pull request please make the title clear and concise, provide a description of the change, and specify any issues that will be closed.
+
+#### Change Files
+Any pull request which includes changes within the `packages/*` directory requires a corresponding change file. Before pushing your changes to create a pull request, be sure you have included the necessary change file(s). To generate a change file, run `yarn change` in the root of the repository. The generated file will be checked into the repo automatically for you as part of the process.
```
Comment (janechu) [inline CONTRIBUTING.md:76]:
```suggestion
Any pull request which includes changes within the `packages/*` directory requires a corresponding change file. Before pushing your changes to create a pull request, be sure you have included the necessary change file(s). To generate a change file, run `npm change` in the root of the repository. The generated file will be checked into the repo automatically for you as part of the process.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+
+:::note
+When working across feature branches, you'll need to target the branch using the following command: `yarn change --branch origin/{branch-name}`.
```
Comment (janechu) [inline CONTRIBUTING.md:79]:
```suggestion
When working across feature branches, you'll need to target the branch using the following command: `npm change --branch origin/{branch-name}`.
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+:::
+
+**Example: Generated change file:**
+```json
+{
+  "type": "minor",
+  "comment": "add fancy new feature for components",
```
Comment (janechu) [inline CONTRIBUTING.md:86]:
```suggestion
  "comment": "Add fancy new feature for components.",
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+  "packageName": "@microsoft/fast-tooling",
+  "email": "name@example.com",
+  "dependentChangeType": "minor",
+  "date": "2021-03-01T19:10:06.323Z"
+}
+```
+
+Running `yarn change` will walk you through a CLI process for generating change files. The process will walk you through selecting the type of change as well as ask you to provide a description of any changes. As a convenience, the utiliity looks to provide recent commit messages for use in the description. *For changes which do not affect the published package(s), please use "none" when selecting the change type*.
```
Comment (janechu) [inline CONTRIBUTING.md:94]:
```suggestion
Running `npm change` will walk you through a CLI process for generating change files. The process will walk you through selecting the type of change as well as ask you to provide a description of any changes. As a convenience, the utility looks to provide recent commit messages for use in the description. *For changes which do not affect the published package(s), please use "none" when selecting the change type*.
```

**Review by janechu — CHANGES_REQUESTED** [review]
I would take a read through https://docs.npmjs.com/cli/v7/using-npm/workspaces to double check my comments as we have yet to implement NPM 7 in any repository but we are planning on attempting it here.

**Review by janechu — APPROVED** [review]

**Pull request merged**